src/address.rs

```diff
@@ -1,8 +1,23 @@
-use crate::keys::Diversifier;
+use crate::keys::{DiversifiedTransmissionKey, Diversifier};
 
 /// A shielded payment address.
+///
+/// # Examples
+///
+/// ```
+/// use orchard::keys::{SpendingKey, FullViewingKey};
+///
+/// let sk = SpendingKey::from_bytes([7; 32]).unwrap();
+/// let address = FullViewingKey::from(&sk).default_address();
+/// ```
 #[derive(Debug)]
 pub struct Address {
     d: Diversifier,
-    pk_d: (),
+    pk_d: DiversifiedTransmissionKey,
+}
+
+impl Address {
```

Comment:
An idea: `From/Into`'s _alongside_ the more descriptive type impl methods defined, where the `From/Into`'s call those, so there is no duplication of logic. I push on this because of the orphan rule, otherwise we could just impl all these `std` traits ourselves: if they aren't defined in here, we'd have to wrap some `orchard` types to achieve them. :/

Reply:
:+1: this is how I like to write typeclass instances in Haskell, and the analogy to traits is apt: write a function that is typed to whatever you're working on, and then use that function to provide for the generalized dispatch via the trait/typeclass.